From 83203ac3bf4fa9edb573e6696b6daabcb337f903 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 12 Nov 2020 23:52:51 +0200
Subject: [PATCH] [SW-12757] gaudi: security: move sram rrs to func

As a preparation for the dynamic security, move the SRAM RRs
initialization to a separate function which will be reused by dynamic
security code.

Change-Id: Ie80353a42a0296e32cfe23a3ff234b5ceecde372
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 04d0b0e..8844336 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1240,6 +1240,22 @@
 	}
 }
 
+static void preboot_secure_hbw_sram(void)
+{
+	/*
+	 * TODO: The below (both SRAM parts) are commented out temporarily
+	 * until the dynamic SRAM RR protection is in place.
+	 */
+	/* SRAM RR 0x7F_F000_0000 - 0x7F_F100_0000 (16MB) */
+/*	preboot_secure_hbw_rr(HBW_RR_SRAM_BGN,
+			      HBW_RR_SRAM_BGN_BASE, HBW_RR_SRAM_BGN_MASK);
+*/
+	/* SRAM RR 0x7F_F100_0000 - 0x7F_F140_0000 (4MB) */
+/*	preboot_secure_hbw_rr(HBW_RR_SRAM_END,
+			      HBW_RR_SRAM_END_BASE, HBW_RR_SRAM_END_MASK);
+*/
+}
+
 #define mmSTM_BASE	0x7FF4000000ULL
 #define mmRAZWI_BASE	0x7FFBFFF000ULL
 
@@ -1248,18 +1264,9 @@
 	/* HBM RR 0 - 32GB */
 	preboot_secure_hbw_rr(HBW_RR_HBM, 0, 0x3FFF800000000);
 
-	/* SRAM RR 0x7F_F000_0000 - 0x7F_F100_0000 (16MB) */
-	/*
-	 * TODO: The below (both SRAM parts) are commented out temporarily
-	 * until the dynamic SRAM RR protection is in place.
-	 */
-/*	preboot_secure_hbw_rr(HBW_RR_SRAM_BGN,
-			      HBW_RR_SRAM_BGN_BASE, HBW_RR_SRAM_BGN_MASK);
-*/
-	/* SRAM RR 0x7F_F100_0000 - 0x7F_F140_0000 (4MB) */
-/*	preboot_secure_hbw_rr(HBW_RR_SRAM_END,
-			      HBW_RR_SRAM_END_BASE, HBW_RR_SRAM_END_MASK);
-*/
+	/* SRAM RR 0x7F_F000_0000 - 0x7F_F140_0000 (20MB) */
+	preboot_secure_hbw_sram();
+
 	/* SRAM Reduct. RR 0x7F_F140_0000 - 0x7F_F180_0000 (4MB) */
 	preboot_secure_hbw_rr(HBW_RR_SRAM_R_BGN, 0x7FF1400000, 0x3FFFFFC00000);
 
